atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -258,9 +333,9 @@ private static SerializableTransactionManager create(
                 transactionManager,
                 kvs,
                 sweepRunner,
-                Suppliers.ofInstance(config.enableSweep()),
-                Suppliers.ofInstance(config.getSweepPauseMillis()),
-                Suppliers.ofInstance(getSweepBatchConfig(config)),
+                () -> MoreObjects.firstNonNull(runtimeConfig.get().enableSweep(), config.enableSweep()),
+                () -> MoreObjects.firstNonNull(runtimeConfig.get().getSweepPauseMillis(), config.getSweepPauseMillis()),
+                () -> getSweepBatchConfig(runtimeConfig.get(), config),
```

Comment:
There is a race condition on the Sweep config here, in that in a single run of `grabLocksAndRun()` you could see `isSweepEnabled`, various `batchConfig`s and a `sweepPauseMillis` from different configs if the config changes are suitably timed.

This might well be benign though. We could fix this by supplying a config instead of the 3 things separately (and having the sweeper take this out and use it as a reference at the beginning of a single run), though that would rely on changes in `BackgroundSweeperImpl`. @gsheasby any thoughts on the safety of this?